### PR TITLE
feat: parse COMMENT

### LINKED ISSUE #1239: Support COMMENT ON (snowflake)
```python
>>> sqlglot.parse_one("COMMENT ON TABLE foo IS 'bar'", read="snowflake")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/oliver.mannion/code3/sqlglot/sqlglot/__init__.py", line 147, in parse_one
    result = dialect.parse(sql, **opts)
  File "/Users/oliver.mannion/code3/sqlglot/sqlglot/dialects/dialect.py", line 163, in parse
    return self.parser(**opts).parse(self.tokenizer.tokenize(sql), sql)
  File "/Users/oliver.mannion/code3/sqlglot/sqlglot/parser.py", line 765, in parse
    return self._parse(
  File "/Users/oliver.mannion/code3/sqlglot/sqlglot/parser.py", line 831, in _parse
    self.raise_error("Invalid expression / Unexpected token")
  File "/Users/oliver.mannion/code3/sqlglot/sqlglot/parser.py", line 874, in raise_error
    raise error
sqlglot.errors.ParseError: Invalid expression / Unexpected token. Line 1, Col: 9.
  COMMENT ON TABLE foo IS 'bar'
```

see https://docs.snowflake.com/en/sql-reference/sql/comment

### PR DESCRIPTION
**Pull request by tekumara**
for [snowflake](https://docs.snowflake.com/en/sql-reference/sql/comment) and [postgres](https://www.postgresql.org/docs/current/sql-comment.html)

resolves #1239

### CODE REVIEW

File: tests/dialects/test_snowflake.py [hunk 1]
```diff
@@ -15,6 +15,7 @@ def test_snowflake(self):
         self.validate_identity(
             'COPY INTO NEW_TABLE ("foo", "bar") FROM (SELECT $1, $2, $3, $4 FROM @%old_table)'
         )
+        self.validate_identity("COMMENT IF EXISTS ON TABLE foo IS 'bar'")
```
Comment (tekumara) [inline tests/dialects/test_snowflake.py:18]: `IF EXISTS` supported by snowflake only I think

File: sqlglot/dialects/snowflake.py [hunk 2]
```diff
@@ -214,6 +214,7 @@ class Tokenizer(tokens.Tokenizer):
 
         KEYWORDS = {
             **tokens.Tokenizer.KEYWORDS,
+            "COMMENT": TokenType.COMMENT,
```
Comment (tobymao) [inline sqlglot/dialects/snowflake.py:217]: this is not needed since you added it to base

File: sqlglot/parser.py [hunk 3]
```diff
@@ -950,6 +951,30 @@ def _retreat(self, index: int) -> None:
     def _parse_command(self) -> exp.Expression:
         return self.expression(exp.Command, this=self._prev.text, expression=self._parse_string())
 
+    def _parse_comment(self, allow_exists=True) -> exp.Expression:
```
Comment (tobymao) [inline sqlglot/parser.py:954]: allow_exists: bool

File: sqlglot/parser.py [hunk 4]
```diff
@@ -950,6 +951,30 @@ def _retreat(self, index: int) -> None:
...
+        start = self._prev
+        exists = self._parse_exists() if allow_exists else False
```
Comment (tobymao) [inline sqlglot/parser.py:956]: probably else None is more consistent

File: sqlglot/parser.py [hunk 5]
```diff
@@ -950,6 +951,30 @@ def _retreat(self, index: int) -> None:
...
+
+        if not self._match(TokenType.ON):
```
Comment (tobymao) [inline sqlglot/parser.py:958]: remove this and go back to COMMENT ON in the tokenizer

File: tests/fixtures/identity.sql [hunk 6]
```diff
@@ -674,7 +674,7 @@ UPDATE db.tbl_name SET foo = 123, foo_1 = 234 WHERE tbl_name.bar = 234
 TRUNCATE TABLE x
 OPTIMIZE TABLE y
 VACUUM FREEZE my_table
-COMMENT ON PROCEDURE my_proc (integer, integer) IS 'Runs a report'
+COMMENT ON PROCEDURE my_proc(integer, integer) IS 'Runs a report'
```
[the remark this answers is not in the document]
Comment (tobymao) [reply tests/fixtures/identity.sql:677]: does this test case test the command portion?
can you add another test on a creatable?

File: sqlglot/parser.py [hunk 7]
```diff
@@ -950,6 +951,30 @@ def _retreat(self, index: int) -> None:
...
+
+        if not kind:
+            return self._parse_as_command(start)
+
+        if kind.token_type in (TokenType.FUNCTION, TokenType.PROCEDURE):
+            this = self._parse_user_defined_function(kind=kind.token_type)
+        else:
+            this = self._parse_id_var()
+
+        if not self._match(TokenType.IS):
```
Comment (tobymao) [inline sqlglot/parser.py:971]: sqlglot is not a validator, so a self._match(TokenType.IS) is sufficient

**Comment by tobymao** [comment]
nice work @tekumara

File: tests/fixtures/identity.sql (hunk 6 above)
[the remark this answers is not in the document]
Comment (tekumara) [reply tests/fixtures/identity.sql:680]: have added `COMMENT ON ACCESS METHOD gin IS 'GIN index access method'` to test the command code path, plus some more creatables, including some paths I missed

File: sqlglot/parser.py (hunk 5 above)
Comment (tekumara) [reply sqlglot/parser.py]: will that work with the `COMMENT [IF EXISTS] ON` [snowflake syntax](https://docs.snowflake.com/en/sql-reference/sql/comment)?

**Review by GeorgeSittas — APPROVED** [review]
Seems legit!

**Pull request merged**